ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualTransactionBuilder.kt

```diff
@@ -0,0 +1,31 @@
+package net.corda.v5.ledger.consensual.transaction
+
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.DigestService
+import net.corda.v5.crypto.merkle.MerkleTreeFactory
+import net.corda.v5.ledger.common.transactions.PrivacySalt
+import net.corda.v5.ledger.consensual.ConsensualState
+import net.corda.v5.ledger.consensual.Party
+import java.security.PublicKey
+import java.security.SecureRandom
+import java.time.Instant
+
+@DoNotImplement
+interface ConsensualTransactionBuilder {
+    val metadata: ConsensualTransactionMetaData?
+    fun withMetadata(metadata: ConsensualTransactionMetaData): ConsensualTransactionBuilder
```

Comment:
I think the meta will all be added internally by the builder, but we should discuss

Reply:
I have modified it according this.